### PR TITLE
refactor(core-data): Error handling for V2 API

### PR DESCRIPTION
**Pull request by cloudxxx8**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [x] The commit subject follows the [Conventional Commits spec](https://github.com/zeke/semantic-pull-requests)
- [x] The commit message follows the [EdgeX Contributor Guide](https://wiki.edgexfoundry.org/display/FA/Contributor%27s+Guide)
- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)
- [x] `make test` has completed successfully

## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [x] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Other... Please describe:

## What is the new behavior?
Modify V2 core-data implementation to adopt the new feature in https://github.com/edgexfoundry/go-mod-core-contracts/pull/279

Issue Number: Fix https://github.com/edgexfoundry/edgex-go/issues/1734

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

### CODE REVIEW

**Comment by cloudxxx8** [comment]
This PR needs to wait until https://github.com/edgexfoundry/go-mod-core-contracts/pull/279 merged

File: internal/pkg/v2/infrastructure/redis/client.go [hunk 1]
```diff
@@ -24,15 +25,13 @@ type Client struct {
 	*redisClient.Client
 }
 
-func NewClient(config db.Configuration, logger logger.LoggingClient) (*Client, error) {
-	var err error
-	dc := &Client{}
+func NewClient(config db.Configuration, logger logger.LoggingClient) (dc *Client, err error) {
```
Comment (lenny-intel) [inline internal/pkg/v2/infrastructure/redis/client.go:28]: Why not returning v2.EdgexError like in event.go?

File: internal/core/data/v2/application/event.go [hunk 2]
```diff
@@ -26,14 +27,14 @@ import (
 
 // The AddEvent function accepts the new event model from the controller functions
 // and invokes addEvent function in the infrastructure layer
-func AddEvent(e models.Event, ctx context.Context, dic *di.Container) (string, error) {
+func AddEvent(e models.Event, ctx context.Context, dic *di.Container) (id string, err v2.EdgexError) {
```
Comment (lenny-intel) [inline internal/core/data/v2/application/event.go:30]: Here you return `v2.EdgexError`, but in other places you left the return type as `error`. Any reason here you use `v2.EdgexError`?

File: internal/pkg/v2/infrastructure/redis/client.go (hunk 1 above)
Comment (cloudxxx8) [reply internal/pkg/v2/infrastructure/redis/client.go:28]: you are right. we should use v2.EdgexError here

Comment (cloudxxx8) [reply internal/pkg/v2/infrastructure/redis/client.go:28]: modified

Comment (lenny-intel) [inline internal/pkg/v2/infrastructure/redis/client.go:28]: Should the return type be `errors.EdgeX`

Comment (cloudxxx8) [reply internal/pkg/v2/infrastructure/redis/client.go]: modified

**Review by lenny-intel — DISMISSED** [review]
LGTM

**Comment by cloudxxx8** [comment]
recheck

**Comment by cloudxxx8** [comment]
rebased

**Review by lenny-intel — APPROVED** [review]
LGTM

**Pull request merged**